folium/_server.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+"""
+Simple HTTP server so we are able to show Maps in browser. Only works with python 3!.
+
+"""
+
+import sys
+if sys.version_info[0] < 3:
+    raise NotImplementedError('Only works with python 3!')
+
+import webbrowser
+from http.server import BaseHTTPRequestHandler, HTTPServer
+from textwrap import dedent
+
+
+# Based on https://stackoverflow.com/a/38945907/3494126
+class TemporaryHTTPServer:
+    """
+    A simple, temporary http web server on the pure Python 3.
+
+    Parameters
+    ----------
+    host: str
+        Local address on which the page will be hosted, default is '127.0.0.1'
+    port: int
+        Corresponding port, default 7000
+    """
+    def __init__(self, host=None, port=None):
+        self.host = host or '127.0.0.1'
+        self.port = port or 7000
+
+        self.server_address = '{host}:{port}'.format(host=self.host, port=self.port)
+        self.full_server_address = 'http://' + self.server_address
+
+    def serve(self, html_data):
+        """
+        Serve html content in a suitable for us manner: allow to gracefully exit using ctrl+c and re-serve some other
+        content on the same host:port
+
+        """
+
+        # we need a request handler with a method `do_GET` which somehow is not provided in the baseline
+        class HTTPServerRequestHandler(BaseHTTPRequestHandler):
+            """
+            An handler of requests for the server, hosting HTML-pages.
+            """
+
+            def do_GET(self):
+                """Handle GET requests"""
+
+                # response from page
+                self.send_response(200)
+
+                # set up headers for pages
+                # content_type = 'text/{0}'.format(page_content_type)
+                self.send_header('Content-type', 'text/html')
+                self.end_headers()
+
+                # writing data on a page
+                self.wfile.write(bytes(html_data, encoding='utf'))
+
+                return
+
+        # create a temporary server
+        HTTPServer.allow_reuse_address = True
+        httpd = HTTPServer((self.host, self.port), HTTPServerRequestHandler)
+
+        # run the temporary http server with graceful exit option
```

Comment:
Most of these comment can be removed. I guess that they were added on SO to explain the answer. We don't need them here.